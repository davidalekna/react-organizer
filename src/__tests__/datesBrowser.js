import 'react-testing-library/cleanup-after-each';
import React from 'react';
import { render } from 'react-testing-library';
import DatesBrowser from '../';

test('should get previews month last weeks offset days', () => {
  const handleStateChange = jest.fn();
  const { getPrevMonthOffset } = setup({
    onStateChange: handleStateChange,
  });
  const result = getPrevMonthOffset({ month: 9, year: 2018 });
  expect(result).toMatchSnapshot();
});

test('should get current month weeks with days and events', () => {
  const handleStateChange = jest.fn();
  const { getCurrentMonth } = setup({
    onStateChange: handleStateChange,
  });
  const result = getCurrentMonth({ month: 9, year: 2018 });
  console.log(result);
});

function setup({ render: renderFn = () => <div />, ...props } = {}) {
  let renderArg;
  const childrenSpy = jest.fn(controllerArg => {
    renderArg = controllerArg;
    return renderFn(controllerArg);
  });
  const utils = render(<DatesBrowser {...props}>{childrenSpy}</DatesBrowser>);
  return { childrenSpy, ...utils, ...renderArg };
}
