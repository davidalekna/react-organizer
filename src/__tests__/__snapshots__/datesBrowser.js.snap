// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get previews month last weeks offset days 1`] = `
Object {
  "days": Array [
    Object {
      "day": 27,
      "events": Array [],
      "name": "Monday",
      "offset": true,
    },
    Object {
      "day": 28,
      "events": Array [],
      "name": "Tuesday",
      "offset": true,
    },
    Object {
      "day": 29,
      "events": Array [],
      "name": "Wednesday",
      "offset": true,
    },
    Object {
      "day": 30,
      "events": Array [],
      "name": "Thursday",
      "offset": true,
    },
    Object {
      "day": 31,
      "events": Array [],
      "name": "Friday",
      "offset": true,
    },
  ],
  "month": 8,
  "name": "August",
  "totalOffsetDays": 5,
}
`;
